programs/server/Server.cpp

```diff
@@ -1069,7 +1069,8 @@ if (ThreadFuzzer::instance().isEffective())
     auto & access_control = global_context->getAccessControl();
     if (config().has("custom_settings_prefixes"))
         access_control.setCustomSettingsPrefixes(config().getString("custom_settings_prefixes"));
-
+    ///set the allow_plaintext_and_no_password setting in context.
+    global_context->setAuthTypeSetting(config().getBool("allow_plaintext_and_no_password",true));
```

Comment:
The Context class has enough complexity already. It's not necessary for Context to be aware of these details. Let's remove changes in `Context` and just call `access_control.setNoPasswordAllowed()` here.